src/sortStudents.ts

```
@@ -1,16 +1,45 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(st: Student):number {
+  return st.grades.reduce((acc, el) => acc + el, 0) / st.grades.length;
+}
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+// eslint-disable-next-line max-len
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder):Student[] {
+  return [...students].sort((a:Student, b:Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(a[sortBy]) - Number(b[sortBy])
+          : Number(b[sortBy]) - Number(a[sortBy]);
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? getAverage(a) - getAverage(b)
+          : getAverage(b) - getAverage(a);
+      default:
```

In the default case, it would be better to throw an error or return a more meaningful value. Returning '0' may lead to unintended consequences.